### DISCUSSION TITLE
Any way to set the condition, If source object is null, the target should also be set as null?

### DISCUSSION THREAD

**Discussion by naeemabbas05** [riok/mapperly#1163] [Q&A]
Is there any option to set the condition, if the source is null, set the target is null. I have tried AllowNullPropertyAssignment, but it is for properties inside source class. I want a condition to be placed on source class itself for e.g. 
```csharp

Class A {
public int Id { get; set; }
...
}

Class B {
public int Id { get;set;}
...
}

[Mapper(AllowNullPropertyAssignment = false)]
public partial class Mapperly : IMapperly
{
        public partial B ToB(A a);
}

```
In the above example, if the A object is null, how to handle this scenario, anyone can help out?

**Comment by latonz** [comment]
Can you provide a little more context? Are these classes / the mapper definition in a nullable context? How do you expect the generated code to look like?

**Reply by naeemabbas05** [reply]
@latonz  for e.g. I am reading a data from API, and among whole set of different classes,  one of the class object is null. How do I use mapperly to convert to another class object. Source class will be null and target class should also be set to null.

**Reply by latonz** [reply]
My question is whether those classes (source, target, mapper) have nullable reference types enabled? See also [MS docs](https://learn.microsoft.com/en-us/dotnet/csharp/nullable-references).
